app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -97,7 +118,10 @@ class DroidCvPattern(private var Mat: Mat? = Mat(), private val OwnsMat: Boolean
                         Template.height
                     )
 
-                    yield(Match(region, score))
+                    val match = Match(region, score)
+
+                    logd("match ($Template on ${this@DroidCvPattern}): $match")
```

Comment:
What does this `this@DroidCvPattern` do?

Reply:
We're inside a lambda passed to the `sequence` function here. So, `this` here would refer to the `Sequence<Match>` object.
So, we have to qualify it with `this@DroidCvPattern` that we want the `DroidCvPattern` instance.

Due to this behaviour, I'm a bit sceptical about using `this::class.simpleName` in place of `[ORIGINAL_CLASS_NAME]::class.simpleName`.